web/src/components/feature/calculation_input/CalculationInput.tsx

```diff
@@ -95,6 +96,18 @@ export const CalculationInput = ({
     </Button>
   )
 
+  const savePackage = (newPackage: TPackage) => {
+    let newPackages = [...packages]
+    if (editablePackage !== undefined) {
+      newPackages = newPackages.filter((x) => x.name !== editablePackage.name)
+    }
```

Comment:
Is this included so that packages can be renamed? Or are there any other reasons to why we need a separate state for editablePackage? Naively I would have thought it is as easy as saying that a package can be edited if it is non-empty.

Reply:
editablePackage does two things. It tells the dialog what mode it is in (edit or new) based on whether editablePackage is undefined or not. If it is in edit mode, ie editablePackage is not undefined, then editablePackage contains the package we want to edit.
I tried finding a better name for this parameter, but could not. If you have any, please tell